pkg/discovery/network/pods.go

```diff
@@ -55,7 +56,7 @@ func findPod(clientSet kubernetes.Interface, labelSelector string) (*v1.Pod, err
 	})
 
 	if err != nil || len(pods.Items) == 0 {
-		return nil, err
+		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", labelSelector)
```

Comment:
if len(pod.Items)==0  we should not return error, in the previous code "err" could still be nil in this case.

Reply:
Sorry not sure I follow. I didn't change the logic. If `len(pod.Items)==0 ` then `err` is nil.

Reply:
@mangelajo if `err` is `nil`, `errors.WithMessagef()` returns `nil` too, so the behaviour doesn’t change here (but the code can be confusing).

Reply:
ahh ok I didn't know that bit @skitt explains thanks.